src/modules/core/components/Fields/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { useField } from 'formik';
+import { PopperProps } from 'react-popper';
+
+import InputLabel from '~core/Fields/InputLabel';
+import Icon from '~core/Icon';
+import { Tooltip } from '~core/Popover';
+
+import styles from './Toggle.css';
+
+const displayName = 'Toggle';
+
+interface Props {
+  name: string;
+  label?: string;
+  disabled?: boolean;
+  tooltipText?: string;
+  elementOnly: boolean;
+  /** Options to pass through the <Popper> element. See here: https://github.com/FezVrasta/react-popper#api-documentation */
+  tooltipPopperProps?: Omit<PopperProps, 'children'>;
+}
+
+const Toggle = ({
+  name,
+  label,
+  disabled,
+  elementOnly,
+  tooltipText,
+  tooltipPopperProps,
+}: Props) => {
+  const [{ onChange, value }] = useField(name);
+
+  return (
+    <div className={styles.container}>
+      {!elementOnly && label && (
+        <InputLabel label={label} appearance={{ colorSchema: 'grey' }} />
+      )}
+      <div>
+        <input
+          name={name}
+          type="checkbox"
+          disabled={disabled}
+          aria-checked={value}
+          aria-disabled={disabled}
+          className={styles.delegate}
+          onChange={onChange}
+        />
+        <span className={disabled ? styles.toggleDisabled : styles.toggle}>
+          <span className={value ? styles.checked : styles.toggleSwitch} />
+        </span>
+      </div>
+      {tooltipText && (
+        <Tooltip
+          appearance={{ theme: 'dark' }}
+          content={tooltipText}
+          trigger="hover"
+          popperProps={tooltipPopperProps}
+        >
+          <div className={styles.icon}>
+            <Icon
+              name="question-mark"
+              appearance={{ size: 'small' }}
+              title=""
+            />
+          </div>
+        </Tooltip>
+      )}
+    </div>
+  );
+};
+
+Toggle.displayName = displayName;
+
+Toggle.defaultProps = {
+  disable: false,
+  elementOnly: false,
+  tooltipPopperProps: {
+    placement: 'top-end',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [-5, 7],
+        },
+      },
+    ],
+  },
+};
```

Comment:
I mean, I can't remember the last time we set `defaultProps` on a react component since it's much easier to just initialize props with what values you need.

Eg:
```jsx
const Toggle = ({
  name,
  label,
  disabled: false,
  elementOnly: false,
  tooltipText,
  tooltipPopperProps = {
    placement: 'top-end',
    modifiers: [
      {
        name: 'offset',
        options: {
          offset: [-5, 7],
        },
      },
    ],
  },
}: Props) => {
  ...
}
```

But in the end it's your choice how you want to implement this, so I'm good with either approach.

Reply:
i followed `Checkbox` example :) but i guess it was an old one. Updated now.